frontends/concrete-python/benchmarks/levenshtein_distance.py

```diff
@@ -0,0 +1,157 @@
+"""
+Benchmarks of the levenshtein distance example.
+"""
+
+from functools import lru_cache
+from pathlib import Path
+from typing import Tuple
+
+import py_progress_tracker as progress
+
+from concrete import fhe
+from examples.levenshtein_distance.levenshtein_distance import Alphabet, LevenshteinDistance
+
+
+@lru_cache
+def levenshtein_fhe(
+    client: fhe.Client,
+    server: fhe.Server,
+    x: Tuple[fhe.Value],
+    y: Tuple[fhe.Value],
+):
+    """
+    Compute levenshtein distance in FHE.
+    """
+
+    if len(x) == 0:
+        return server.run(
+            len(y),
+            function_name="constant",
+            evaluation_keys=client.evaluation_keys,
+        )
+
+    if len(y) == 0:
+        return server.run(
+            len(x),
+            function_name="constant",
+            evaluation_keys=client.evaluation_keys,
+        )
+
+    if_equal = levenshtein_fhe(client, server, x[1:], y[1:])
+    case_1 = levenshtein_fhe(client, server, x[1:], y)
+    case_2 = levenshtein_fhe(client, server, x, y[1:])
+    case_3 = if_equal
+
+    is_equal = server.run(
+        x[0],
+        y[0],
+        function_name="equal",
+        evaluation_keys=client.evaluation_keys,
+    )
+    result = server.run(
+        is_equal,
+        if_equal,
+        case_1,
+        case_2,
+        case_3,
+        function_name="mix",
+        evaluation_keys=client.evaluation_keys,
+    )
+
+    return result
+
+
+def targets():
+    """
+    Generates targets to benchmark.
+    """
+
+    result = []
+    for alphabet in ["ACTG", "string"]:
```

Comment:
not the other ones?

Reply:
> not the other ones?

Just like testing, we have a limited time to complete the benchmarks (it was 6 hours I believe). And this shows us two differently sized alphabets, which is enough to do basic comparison.

Reply:
I would be in favor of at least a few tests for each alphabet
Which is even more important to me, if you want to remove the --autotest options from the .py (which you said in another comment)

Reply:
This is for benchmarks, not tests. Do we really need to benchmark uppercase and lowercase to obtain the same results?

Reply:
ok you're right, my bad sorry! maybe I would be considering adding the slowest alphabet (sTrInG or StRiNg, I don't remind), since it's where we could see the best improvements over time, but nothing blocking for me.

Reply:
We can, but the problem is that it increases the execution time exponentially :sweat: Let's start benchmarking like this and we can add new benchmarks if we have the time budget later. Okay for you?

Reply:
ok (but it means we'll never)

Reply:
Created https://github.com/zama-ai/concrete-internal/issues/844 so that's not the case :slightly_smiling_face: